File: archive/logic/TuringMachineSimulationNew/TuringMachineConfig.cs
using TuringMachineSimulation.Logic.States.Transition;
using TuringMachineSimulation.Logic.Tapes;
using TuringMachineSimulation.Logic.Heads;

namespace TuringMachineSimulation.Logic
{
    public class TuringMachineConfig
    {
        public int NumberOfHeads { get; set; }
        public ICollection<HeadType> HeadTypes { get; set; } = new List<HeadType>();
        public ICollection<int> InitialPositions { get; set; } = new List<int>();
        public ICollection<int> TapesReference { get; set; } = new List<int>();
        public ICollection<TransitionNode> TransitionNodes { get; set; } = new List<TransitionNode>();
        public ICollection<TransitionStatement> Transitions { get; set; } = new List<TransitionStatement>();
        public TransitionNode StartNode { get; set; }

        public TuringMachineConfig(
            int numberOfHeads, 
            ICollection<HeadType> headTypes, 
            ICollection<int> initialPositions, 
            ICollection<int> tapesReference, 
            ICollection<TransitionNode> transitionNodes,
            ICollection<TransitionStatement> transitions,
            TransitionNode startNode)
        {
            NumberOfHeads = numberOfHeads;
            HeadTypes = headTypes;
            InitialPositions = initialPositions;
            TapesReference = tapesReference;
            TransitionNodes = transitionNodes;
            Transitions = transitions;
            StartNode = startNode;
        }
    }
}

File: frontend/turing-machine-web/src/logic/TuringMachineConfig.ts
import { HeadTypes } from "./Heads/HeadTypes";
import { TransitionNode } from "./States/Transitions/TransitionNode"
import { TransitionStatement } from "./States/Transitions/TransitionStatement"

export class TuringMachineConfig {
    public NumberOfHeads: number;
    public HeadTypes: HeadTypes[];
    public InitialPositions: number[];
    public TapesReference: number[];
    public TransitionNodes: TransitionNode[];
    public Statements: TransitionStatement[];
    public StartNode: TransitionNode;

    /**
     * Constructor for creating a machine configuration.
     */
    constructor(
        numberOfHeads: number,
        headTypes: HeadTypes[],
        initialPositions: number[],
        tapesReference: number[],
        transitionNodes: TransitionNode[],
        statements: TransitionStatement[],
        startNode: TransitionNode
    ) {
        this.NumberOfHeads = numberOfHeads;
        this.HeadTypes = headTypes;
        this.InitialPositions = initialPositions;
        this.TapesReference = tapesReference;
        this.TransitionNodes = transitionNodes;
        this.Statements = statements;
        this.StartNode = startNode;
    }
}